neuralpredictors/layers/shifters/static_affine.py

```diff
@@ -21,6 +21,8 @@ def __init__(self, input_channels=2, output_channels=2, bias=True):
         super().__init__(input_channels, output_channels, bias=bias)
 
     def forward(self, x, trial_idx=None):
+        if trial_idx is not None:
+            raise ValueError("This shifter network does not support trial indexing. Set trial_idx to None")
```

Comment:
I am not so sure about the error here. I think it's unnecessary since the user would probably choose a specific shifter depending on whether they want to make use of `trial_index` or not. Having `trial_index` as input argument for all shifters is just a convenience and a convention (just like the `shift` argument for the readouts). Also, the error has potential to become a hassle later on. Here is an example:
Imagine we have two models, where each one of them use a different shifter (one that uses `trial_index`, one doesn't). With this error, we would need two dataloaders (one that returns `trial_idx` and one that does not). Without it, we can just share one dataloader between the two models and have them both work smoothly (at least as far as the shifter is concerned)

Reply:
To summarize this issue as we talked about it in the code-review session: I agree that an error or warning is over the top, and I'd suggest to use **kwargs in the shifter instead.

Reply:
In the code review meeting today we decided to raise a warning here if `trial_idx is not None`.